<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="" />
  <title>Analiza metylacji</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="index_files/reveal.js-3.3.0/css/reveal.css"/>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="index_files/reveal.js-3.3.0/css/theme/black.css" id="theme">

<style type="text/css">
.reveal section img {
  background: rgba(255, 255, 255, 0.85);
}
</style>

  <!-- some tweaks to reveal css -->
  <style type="text/css">
    .reveal h1 { font-size: 2.0em; }
    .reveal h2 { font-size: 1.5em;  }
    .reveal h3 { font-size: 1.25em;	}
    .reveal h4 { font-size: 1em;	}

    .reveal .slides>section,
    .reveal .slides>section>section {
      padding: 0px 0px;
    }



    .reveal table {
      border-width: 1px;
      border-spacing: 2px;
      border-style: dotted;
      border-color: gray;
      border-collapse: collapse;
      font-size: 0.7em;
    }

    .reveal table th {
      border-width: 1px;
      padding-left: 10px;
      padding-right: 25px;
      font-weight: bold;
      border-style: dotted;
      border-color: gray;
    }

    .reveal table td {
      border-width: 1px;
      padding-left: 10px;
      padding-right: 25px;
      border-style: dotted;
      border-color: gray;
    }

  </style>

    <style type="text/css">code{white-space: pre;}</style>

    <!-- Printing and PDF exports -->
    <script>
      var link = document.createElement( 'link' );
      link.rel = 'stylesheet';
      link.type = 'text/css';
      link.href = window.location.search.match( /print-pdf/gi ) ? 'index_files/reveal.js-3.3.0/css/print/pdf.css' : 'index_files/reveal.js-3.3.0/css/print/paper.css';
      document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>
    <!--[if lt IE 9]>
    <script src="index_files/reveal.js-3.3.0/lib/js/html5shiv.js"></script>
    <![endif]-->

    <link href="index_files/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section data-background-image="imgs/bg-small.png">
    <h1 class="title">Analiza metylacji</h1>
  <h1 class="subtitle"><small> <br>Aleksandra Brodecka   <a href="http://r-addict.com/About.html">Marcin Kosiński</a> </small></h1>
    <h2 class="author"><small><a href='https://github.com/abrodecka'><i class='fa fa-github'></i></a>  <a href='mailto:aleksandrabrodecka@gmail.com'><i class='fa fa-envelope-o'></i></a>                                            <a href='https://r-addict.com'><i class='fa fa-comment'></i></a>  <a href='https://stackoverflow.com/users/3857701'><i class='fa fa-stack-overflow'></i></a>  <a href='https://github.com/MarcinKosinski'><i class='fa fa-github'></i></a>  <a href='mailto:m.p.kosinski@gmail.com'><i class='fa fa-envelope-o'></i></a></small><br></h2>
    <h3 class="date">12 Grudnia, 2016</h3>
</section>

<section><section id="plan-prezentacji" class="titleslide slide level1"><h1>Plan prezentacji</h1></section><section id="plan-prezentacji-1" class="slide level2">
<h1>Plan prezentacji</h1>
<small>
<ul>
<li>Cel analizy</li>
<li>Format danych (raw i dostosowane do methylKit)</li>
<li>methylKit - podstawy</li>
<li>methylKit - wykresy</li>
<li>analiza docelowa
<ul>
<li>na regionach długości tysiąc</li>
<li>na regionach odpowiadających genom</li>
</ul></li>
<li>Prezentacja aplikacji</li>
</ul>
</small>
</section></section>
<section><section id="cel-analizy" class="titleslide slide level1"><h1>Cel analizy</h1></section><section id="cel-analizy-1" class="slide level2">
<h1>Cel analizy</h1>
<p>Znalezienie regionów o zróżniocowanym stopniu metylacji (<em>ang. Differentially methylated regions</em>).</p>
<p>Współpraca z prof. Płoskim <br>(Zakład Genetyki Medycznej WUM).</p>
</section><section id="przykad-analizy" class="slide level2">
<h1>Przykład analizy</h1>
<small>
<p>W badaniu opublikowanym w 2013 roku dokonano porównania stopnia metylacji u kobiet z osteoporozą i kobiet z chorobą zwyrodnieniową stawów. Odkryto około 250 regionów o zróżnicowanym stopniu metylacji. Były to głównie regiony kodujące czynniki transkrypcyjne, które biorą udział w różnicowaniu komórek i tworzenie szkieletu, co sugeruje, że stopień metylacji może istotny przy tworzeniu tych zaburzeń.</p>
</small>
</section></section>
<section><section id="postac-danych" class="titleslide slide level1"><h1>Postać danych</h1></section><section id="postac-danych-1" class="slide level2">
<h1>Postać danych</h1>
<small>
<p>Dwie próbki pochodzące od jednej pacjentki <br> EK - komórka nie poddana chorobie - ponad 6mln rekordów, <br>EU - komórka poddana chorobie - 11 mln rekordów.</p>
<ul>
<li>postać danych przed tranformacją:</li>
</ul>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">CHROM</th>
<th style="text-align: center;">POS</th>
<th style="text-align: right;">REF</th>
<th style="text-align: right;">ALT</th>
<th style="text-align: right;">DP</th>
<th style="text-align: right;">AF</th>
<th style="text-align: right;">meth</th>
<th style="text-align: right;">unmeth</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">chrM</td>
<td style="text-align: center;">106</td>
<td style="text-align: right;">G</td>
<td style="text-align: right;">C</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">chr1</td>
<td style="text-align: center;">2198933</td>
<td style="text-align: right;">C</td>
<td style="text-align: right;">G</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">0.3478</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">chrY</td>
<td style="text-align: center;">9269317</td>
<td style="text-align: right;">C</td>
<td style="text-align: right;">G</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
</tr>
</tbody>
</table>
<ul>
<li>postać danych po transformacji:</li>
</ul>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">chr</th>
<th style="text-align: center;">start</th>
<th style="text-align: right;">end</th>
<th style="text-align: right;">coverage.EK</th>
<th style="text-align: right;">met.EK</th>
<th style="text-align: right;">coverage.EU</th>
<th style="text-align: right;">met.EU</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">chr1</td>
<td style="text-align: center;">15513</td>
<td style="text-align: right;">15513</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">1.00</td>
</tr>
<tr class="even">
<td style="text-align: left;">chrY</td>
<td style="text-align: center;">59033091</td>
<td style="text-align: right;">59033091</td>
<td style="text-align: right;">99</td>
<td style="text-align: right;">0.67</td>
<td style="text-align: right;">166</td>
<td style="text-align: right;">0.77</td>
</tr>
<tr class="odd">
<td style="text-align: left;">chr20</td>
<td style="text-align: center;">3139367</td>
<td style="text-align: right;">3139367</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">0.75</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">0.76</td>
</tr>
</tbody>
</table>
<ul>
<li>jedna próbka - ponad 2 mln rekordów</li>
</ul>
</small>
</section></section>
<section><section id="pakiet-methylkit" class="titleslide slide level1"><h1>Pakiet methylKit</h1></section><section id="podstawowe-funkcje" class="slide level2">
<h1>Podstawowe funkcje</h1>
<p>Wczytywanie danych: (muszą być specjalniej postaci! )</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">file.list &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="st">&quot;Sample_40092EK.tab&quot;</span>,
                  <span class="st">&quot;Sample_40092EU.tab&quot;</span>)
myobj &lt;-<span class="st"> </span><span class="kw">methRead</span>(file.list, <span class="dt">sample.id =</span> <span class="kw">list</span>(<span class="st">&quot;EK&quot;</span>, <span class="st">&quot;EU&quot;</span>),
                  <span class="dt">assembly =</span> <span class="st">&quot;hg18&quot;</span>, <span class="dt">treatment =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">0</span>), 
                  <span class="dt">context =</span> <span class="st">&quot;CpG&quot;</span>)</code></pre></div>
<p>Łączenie próbek:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">meth &lt;-<span class="st"> </span><span class="kw">unite</span>(myobj)</code></pre></div>
</section></section>
<section><section id="analiza-regiony-o-zroznicowanej-metylacji" class="titleslide slide level1"><h1>Analiza <br> Regiony o zróżnicowanej metylacji</h1></section><section id="okna-o-dugosci-tysiac" class="slide level2">
<h1>Okna o długości tysiąc</h1>
<small>
<ol type="1">
<li><p>Podział danych na regiony. Każda grupa <span class="math inline">\((chr, tiles)\)</span> spełnia warunki: <span class="math display">\[chromosom = chr, start &gt;= tiles * 1000, end &lt;= (tiles + 1)*1000\]</span></p></li>
<li><p>Dla każdej grupy wykonuję test Wilcoxona dla par obserwacji:</p></li>
</ol>
<ul>
<li><span class="math inline">\(d_i = met_{EK_i} - met_{EU_i}\)</span> (iid, symetryczne względem mediany = <span class="math inline">\(\theta\)</span>)</li>
<li>rangowanie <span class="math inline">\(R_i\)</span> zbioru: <span class="math inline">\(|d_1|, \ldots |d_n|\)</span></li>
<li>statystyka: <span class="math inline">\(W^+\)</span> suma <span class="math inline">\(R_i\)</span>, dla których <span class="math inline">\(d_i &gt; 0\)</span> (przybliżana r. normalnym)</li>
<li><span class="math inline">\(H_0: \theta = 0\)</span></li>
<li><span class="math inline">\(H_1: \theta \neq 0\)</span></li>
</ul>
</small>
</section><section id="okna-o-dugosci-tysiac-linki" class="slide level2">
<h1>Okna o długości tysiąc <br> Linki</h1>
<small>
<p>Przykładowe wyniki:</p>
<ul>
<li>chr13: <a href="http://127.0.0.1:4444/?chrom=chr13&amp;s=58207000&amp;e=58208000" class="uri">http://127.0.0.1:4444/?chrom=chr13&amp;s=58207000&amp;e=58208000</a></li>
<li>chr22: <a href="http://127.0.0.1:4444/?chrom=chr22&amp;s=38610000&amp;e=38611000" class="uri">http://127.0.0.1:4444/?chrom=chr22&amp;s=38610000&amp;e=38611000</a></li>
<li>chr2: <a href="http://127.0.0.1:4444/?chrom=chr2&amp;s=171573000&amp;e=171574000" class="uri">http://127.0.0.1:4444/?chrom=chr2&amp;s=171573000&amp;e=171574000</a></li>
<li>chr22: <a href="http://127.0.0.1:4444/?chrom=chr22&amp;s=29876000&amp;e=29877000" class="uri">http://127.0.0.1:4444/?chrom=chr22&amp;s=29876000&amp;e=29877000</a></li>
<li>chr5: <a href="http://127.0.0.1:4444/?chrom=chr5&amp;s=202900000&amp;e=202901000" class="uri">http://127.0.0.1:4444/?chrom=chr5&amp;s=202900000&amp;e=202901000</a></li>
</ul>
</small>
</section></section>
<section><section id="analiza-regiony-odpowiadajace-genom" class="titleslide slide level1"><h1>Analiza <br> Regiony odpowiadające genom</h1></section><section id="jak-znalezc-regiony-odpowiadajace-genom" class="slide level2">
<h1>Jak znaleźć regiony odpowiadające genom?</h1>
<p><small> <a href="https://www.researchgate.net/post/How_can_I_get_the_chromosomal_location_of_a_list_of_genes"><i class='fa fa-stack-overflow'></i> How can I get the chromosomal location of a list of genes?</a> -&gt; bioMart</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(biomaRt)
ensembl &lt;-<span class="st"> </span><span class="kw">useMart</span>(<span class="st">&quot;ensembl&quot;</span>,
                   <span class="dt">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span>)
<span class="co"># head(listAttributes(ensembl),50)</span>
attributes &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;chromosome_name&#39;</span>, <span class="st">&#39;start_position&#39;</span>, <span class="st">&#39;end_position&#39;</span>,
                <span class="st">&#39;gene_biotype&#39;</span>, <span class="st">&#39;hgnc_id&#39;</span>, <span class="st">&#39;hgnc_symbol&#39;</span>)
<span class="co"># head(listFilters(ensembl),20)</span>
genes &lt;-<span class="st"> </span><span class="kw">getBM</span>(<span class="dt">attributes =</span> attributes, <span class="dt">mart =</span> ensembl, 
               <span class="dt">filter =</span> <span class="kw">c</span>(<span class="st">&#39;chromosome_name&#39;</span>), <span class="dt">value =</span> <span class="kw">c</span>(<span class="dv">1</span>:<span class="dv">22</span>,<span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>, <span class="st">&quot;M&quot;</span>))
genes_hgnc &lt;-<span class="st"> </span>dplyr::<span class="kw">filter</span>(genes, hgnc_symbol !=<span class="st"> &quot;&quot;</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">chromosome_name</th>
<th style="text-align: right;">start_position</th>
<th style="text-align: right;">end_position</th>
<th style="text-align: left;">gene_biotype</th>
<th style="text-align: left;">hgnc_id</th>
<th style="text-align: left;">hgnc_symbol</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: right;">85762559</td>
<td style="text-align: right;">85763138</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">HGNC:36410</td>
<td style="text-align: left;">RPS2P25</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">222503883</td>
<td style="text-align: right;">222504843</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">HGNC:38555</td>
<td style="text-align: left;">GAPDHP49</td>
</tr>
<tr class="odd">
<td style="text-align: left;">8</td>
<td style="text-align: right;">81807772</td>
<td style="text-align: right;">81808726</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">HGNC:48765</td>
<td style="text-align: left;">HNRNPA1P36</td>
</tr>
<tr class="even">
<td style="text-align: left;">9</td>
<td style="text-align: right;">96928310</td>
<td style="text-align: right;">96940253</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">HGNC:23449</td>
<td style="text-align: left;">NUTM2G</td>
</tr>
<tr class="odd">
<td style="text-align: left;">13</td>
<td style="text-align: right;">22696023</td>
<td style="text-align: right;">22696574</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">HGNC:3994</td>
<td style="text-align: left;">FTH1P7</td>
</tr>
<tr class="even">
<td style="text-align: left;">8</td>
<td style="text-align: right;">122769274</td>
<td style="text-align: right;">122770150</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">HGNC:21261</td>
<td style="text-align: left;">CDK5P1</td>
</tr>
</tbody>
</table>
</small>
</section><section id="jak-znalezc-pokrycia-odczytow-w-genach" class="slide level2">
<h1>Jak znaleźć pokrycia odczytów w genach?</h1>
<small>
<p><a href="http://stackoverflow.com/questions/3916195/finding-overlap-in-ranges-with-r"><i class='fa fa-stack-overflow'></i> how to find interval overlaps in r -&gt; Finding overlap in ranges with R</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">overlaps_find_1 &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(
  <span class="kw">as</span>(analysis1[[<span class="dv">1</span>]], <span class="st">&quot;GRanges&quot;</span>), <span class="co"># myobj[[1]]</span>
  <span class="kw">as</span>(genes_hgnc, <span class="st">&quot;GRanges&quot;</span>),
  <span class="dt">type =</span> <span class="st">&quot;within&quot;</span>)
<span class="kw">cbind</span>(
  genes_hgnc[<span class="kw">subjectHits</span>(overlaps_find_1), ],
  <span class="kw">as.data.frame</span>(
    <span class="kw">as</span>(analysis1[[<span class="dv">1</span>]], <span class="st">&quot;GRanges&quot;</span>)
                )[<span class="kw">queryHits</span>(overlaps_find_1), <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">6</span>:<span class="dv">8</span>)] %&gt;%
<span class="st">    </span>dplyr::<span class="kw">rename</span>(<span class="dt">base =</span> start) %&gt;%
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">group =</span> <span class="st">&quot;EK&quot;</span>)
  ) -&gt;<span class="st"> </span>EK</code></pre></div>
<p><a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.pdf">An Introduction to the GenomicRanges Package</a></p>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">seqnames</th>
<th style="text-align: right;">start</th>
<th style="text-align: right;">end</th>
<th style="text-align: left;">gene_biotype</th>
<th style="text-align: left;">hgnc_id</th>
<th style="text-align: left;">hgnc_symbol</th>
<th style="text-align: right;">base</th>
<th style="text-align: right;">coverage</th>
<th style="text-align: right;">numCs</th>
<th style="text-align: right;">numTs</th>
<th style="text-align: left;">group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">825138</td>
<td style="text-align: right;">859446</td>
<td style="text-align: left;">processed_transcript</td>
<td style="text-align: left;">HGNC:49377</td>
<td style="text-align: left;">LINC01128</td>
<td style="text-align: right;">825448</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">EK</td>
</tr>
<tr class="even">
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">825138</td>
<td style="text-align: right;">859446</td>
<td style="text-align: left;">processed_transcript</td>
<td style="text-align: left;">HGNC:49377</td>
<td style="text-align: left;">LINC01128</td>
<td style="text-align: right;">825449</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">13</td>
<td style="text-align: left;">EK</td>
</tr>
<tr class="odd">
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">825138</td>
<td style="text-align: right;">859446</td>
<td style="text-align: left;">processed_transcript</td>
<td style="text-align: left;">HGNC:49377</td>
<td style="text-align: left;">LINC01128</td>
<td style="text-align: right;">825626</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">EK</td>
</tr>
<tr class="even">
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">825138</td>
<td style="text-align: right;">859446</td>
<td style="text-align: left;">processed_transcript</td>
<td style="text-align: left;">HGNC:49377</td>
<td style="text-align: left;">LINC01128</td>
<td style="text-align: right;">825627</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">EK</td>
</tr>
<tr class="odd">
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">825138</td>
<td style="text-align: right;">859446</td>
<td style="text-align: left;">processed_transcript</td>
<td style="text-align: left;">HGNC:49377</td>
<td style="text-align: left;">LINC01128</td>
<td style="text-align: right;">825684</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">8</td>
<td style="text-align: left;">EK</td>
</tr>
<tr class="even">
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">825138</td>
<td style="text-align: right;">859446</td>
<td style="text-align: left;">processed_transcript</td>
<td style="text-align: left;">HGNC:49377</td>
<td style="text-align: left;">LINC01128</td>
<td style="text-align: right;">825685</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">EK</td>
</tr>
</tbody>
</table>
</small>
</section><section id="jak-znalezc-zroznicowane-geny" class="slide level2">
<h1>Jak znaleźć zróżnicowane geny?</h1>
<small>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">EK_EU %&gt;%<span class="st"> </span><span class="co"># dplyr::bind_rows(EK, EU)</span>
<span class="st">  </span><span class="kw">filter</span>(hgnc_symbol %in%<span class="st"> </span>genes_names) %&gt;%<span class="st"> </span><span class="co"># &gt; 100 odczytow w EK i EU</span>
<span class="st">  </span><span class="kw">group_by</span>(hgnc_symbol) %&gt;%
<span class="st">    </span><span class="kw">do</span>(<span class="dt">mod =</span> <span class="kw">glm</span>(<span class="kw">cbind</span>(numCs, numTs) ~<span class="st"> </span>group+<span class="kw">as.factor</span>(base), <span class="dt">data =</span>.,
          <span class="dt">family =</span> <span class="st">&quot;binomial&quot;</span>)) -&gt;<span class="st"> </span>models
models %&gt;%<span class="st"> </span><span class="kw">do</span>(<span class="kw">data.frame</span>(
                <span class="dt">var =</span> <span class="kw">names</span>(<span class="kw">coef</span>(.$mod)),
                <span class="kw">coef</span>(<span class="kw">summary</span>(.$mod)))) %&gt;%
<span class="st">  </span><span class="kw">filter</span>(var ==<span class="st"> &quot;groupEU&quot;</span>) -&gt;<span class="st"> </span>res_do
<span class="co"># + troche brudnego przerabiania</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th style="text-align: right;">Estimate</th>
<th style="text-align: right;">pval</th>
<th style="text-align: left;">hgnc_symbol</th>
<th style="text-align: left;">chromosome_name</th>
<th style="text-align: right;">start_position</th>
<th style="text-align: right;">end_position</th>
<th style="text-align: left;">gene_biotype</th>
<th style="text-align: left;">hgnc_id</th>
<th style="text-align: right;">pval.adj</th>
<th style="text-align: right;">EK_reads_in_gene</th>
<th style="text-align: right;">EU_reads_in_gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">0.8717109</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">CYSTM1</td>
<td style="text-align: left;">5</td>
<td style="text-align: right;">140174642</td>
<td style="text-align: right;">140282052</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">HGNC:30239</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">752</td>
<td style="text-align: right;">952</td>
</tr>
<tr class="even">
<td style="text-align: right;">1.7999432</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">PIK3R5</td>
<td style="text-align: left;">17</td>
<td style="text-align: right;">8878911</td>
<td style="text-align: right;">8965712</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">HGNC:30035</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">253</td>
<td style="text-align: right;">281</td>
</tr>
<tr class="odd">
<td style="text-align: right;">1.2661603</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">LINC01121</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">45164870</td>
<td style="text-align: right;">45323295</td>
<td style="text-align: left;">processed_transcript</td>
<td style="text-align: left;">HGNC:49266</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">513</td>
<td style="text-align: right;">637</td>
</tr>
<tr class="even">
<td style="text-align: right;">2.0360239</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">ZNF385C</td>
<td style="text-align: left;">17</td>
<td style="text-align: right;">42025576</td>
<td style="text-align: right;">42098479</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">HGNC:33722</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">260</td>
<td style="text-align: right;">293</td>
</tr>
<tr class="odd">
<td style="text-align: right;">-0.4512390</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">PRDM16</td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">3069168</td>
<td style="text-align: right;">3438621</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">HGNC:14000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2766</td>
<td style="text-align: right;">3162</td>
</tr>
<tr class="even">
<td style="text-align: right;">2.1870519</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">CYP2F1</td>
<td style="text-align: left;">19</td>
<td style="text-align: right;">41114432</td>
<td style="text-align: right;">41128366</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">HGNC:2632</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">143</td>
</tr>
</tbody>
</table>
</small>
</section></section>
    </div>
  </div>

  <script src="index_files/reveal.js-3.3.0/lib/js/head.min.js"></script>
  <script src="index_files/reveal.js-3.3.0/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display the page number of the current slide
        slideNumber: true,
        // Push each slide change to the browser history
        history: true,
        // Vertical centering of slides
        center: true,
        // Transition style
        transition: 'default', // none/fade/slide/convex/concave/zoom
        // Transition style for full page slide backgrounds
        backgroundTransition: 'default', // none/fade/slide/convex/concave/zoom

        // Optional reveal.js plugins
        dependencies: [
        ]
      });
    </script>
  <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
      var script = document.createElement("script");
      script.type = "text/javascript";
      script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
      document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>

<script>
  (function() {
    if (window.jQuery) {
      Reveal.addEventListener( 'slidechanged', function(event) {  
        window.jQuery(event.previousSlide).trigger('hidden');
        window.jQuery(event.currentSlide).trigger('shown');
      });
    }
  })();
</script>


  </body>
</html>
